stl/src/filesys.cpp

```diff
@@ -111,10 +111,9 @@ _FS_DLL void* __CLRCALL_PURE_OR_CDECL _Open_dir(
         _Wildname.append(L"\\*");
     }
 
-    void* _Handle =
-        FindFirstFileExW(_Wildname.c_str(), FindExInfoStandard, &_Dentry, FindExSearchNameMatch, nullptr, 0);
+    void* _Handle = FindFirstFileW(_Wildname.c_str(), &_Dentry);
     if (_Handle == INVALID_HANDLE_VALUE) { // report failure
-        _Errno = ERROR_BAD_PATHNAME;
+        _Errno = GetLastError();

```

Comment:
This part is a more severe change than the API change, yet not even mentioned in PR description. Possibly it even deserves a separate PR.